mtp_api/apps/performance/views.py

```diff
@@ -444,3 +450,69 @@ def get_ticket_count(self, env_tag, app_tag, start_date, end_date):
             return count
         except requests.RequestException:
             return 0
+
+
+class PerformanceDataView(ListAPIView):
+    serializer_class = PerformanceDataSerializer
+    pagination_class = None
+
+    permission_classes = (
+        IsAuthenticated, SendMoneyClientIDPermissions,
+    )
+
+    def get_queryset(self):
+        today = timezone.localdate()
+        a_year_ago = today - datetime.timedelta(weeks=52)
+
+        filters = {
```

Comment:
if i recall correctly, we're not using django-filters filter sets because there was no way to specify fallback values when none were provided, right?

i know this is an internal-only api but i'm not a huge fan of the very-limited validation of `week__gte` and `week__lt`. it shouldn't be possible to inject security problems, but it'd be easy to cause a harmless 500 error.

i wonder if it'd be possible (unless it's too much hassle!) to add a filter set and have these defaults set on the request params somehow. maybe something like `self.request.query_params.setdefault(…)` overriding a method such as `filter_queryset`..?

Reply:
It's an internal API but I guess it's a fair point. I'm not really familiar with all these filters but I'll play with alternatives to make this more strict.

Reply:
Hey Igor, this is what I came up with: https://github.com/ministryofjustice/money-to-prisoners-api/pull/669/commits/8d30b1941da5bc2c811632898977bed7f1ba6f46

I think it's not bad although there may of course another million ways of doing this as usual :)

Hope it makes sense and it's addressing the issue you had in mind, but we can have a chat about this tomorrow.